src/github.com/getlantern/enproxy/proxy.go

```diff
@@ -348,11 +348,15 @@ func clientIpFor(req *http.Request) string {
 }
 
 func badGateway(resp http.ResponseWriter, msg string) {
-	log.Errorf("Responding Bad Gateway: %s", msg)
-	resp.WriteHeader(http.StatusBadGateway)
+	respond(http.StatusBadGateway, resp, fmt.Sprintf("Enproxy server responding Bad Gateway: %s", msg))
 }
 
 func forbidden(resp http.ResponseWriter, msg string) {
-	log.Errorf("Responding Forbidden: %s", msg)
-	resp.WriteHeader(http.StatusForbidden)
+	respond(http.StatusForbidden, resp, fmt.Sprintf("Enproxy server responding Forbidden: %s", msg))
+}
+
+func respond(status int, resp http.ResponseWriter, msg string) {
+	log.Errorf(msg)
+	resp.WriteHeader(status)
+	resp.Write([]byte(msg))
```

Comment:
Right yeah that's the idea @fffw. I added logging on the client side separately. I think those response bodies will actually show a CloudFlare message body, but this will allow us to be sure it's not the 403 we're returning ourselves.